From b6032f4f388fc81f0aaa05c24c0564be88683fc3 Mon Sep 17 00:00:00 2001
From: Ren Jianing <jianing.ren@rock-chips.com>
Date: Wed, 5 Aug 2020 16:47:30 +0800
Subject: [PATCH] phy: phy-rockchip-usb: add USB external connectors

This patch add EXTCON_USB, EXTCON_USB_HOST and EXTCON_USB_VBUS_EN
to rockchip_usb_phy_extcon_cable. Without these extcons, the probe
of battery charger(such as rk818_charger) will fail when registering
extcon notifier.

Signed-off-by: Ren Jianing <jianing.ren@rock-chips.com>
Change-Id: I17774f63bc84fc1f5afa11d2ecc9d0eb2469acbc
---
 drivers/phy/rockchip/phy-rockchip-usb.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-usb.c b/drivers/phy/rockchip/phy-rockchip-usb.c
index 3591c0ac5a16..f3a391fcbc77 100644
--- a/drivers/phy/rockchip/phy-rockchip-usb.c
+++ b/drivers/phy/rockchip/phy-rockchip-usb.c
@@ -93,6 +93,9 @@ enum usb_chg_state {
 };
 
 static const unsigned int rockchip_usb_phy_extcon_cable[] = {
+	EXTCON_USB,
+	EXTCON_USB_HOST,
+	EXTCON_USB_VBUS_EN,
 	EXTCON_CHG_USB_SDP,
 	EXTCON_CHG_USB_CDP,
 	EXTCON_CHG_USB_DCP,
-- 
2.35.3

